Gem/Code/Source/ProceduralSkinnedMeshUtils.cpp

```diff
@@ -81,28 +157,59 @@ namespace AtomSampleViewer
-        modelLodCreator.AddMeshStreamBuffer(RHI::ShaderSemantic{ "NORMAL" }, AZ::Name(), AZ::RPI::BufferAssetView{ normalBuffer, normalBuffer->GetBufferViewDescriptor() });
-        modelLodCreator.AddMeshStreamBuffer(RHI::ShaderSemantic{ "TANGENT" }, AZ::Name(), AZ::RPI::BufferAssetView{ tangentBuffer, tangentBuffer->GetBufferViewDescriptor() });
-        modelLodCreator.AddMeshStreamBuffer(RHI::ShaderSemantic{ "BITANGENT" }, AZ::Name(), AZ::RPI::BufferAssetView{ bitangentBuffer, bitangentBuffer->GetBufferViewDescriptor() });
-        modelLodCreator.AddMeshStreamBuffer(RHI::ShaderSemantic{ "UV" }, AZ::Name(), AZ::RPI::BufferAssetView{ uvBuffer, uvBuffer->GetBufferViewDescriptor() });
-
-        AZ::Aabb localAabb = AZ::Aabb::CreateCenterHalfExtents(AZ::Vector3(0.0f, 0.0f, 0.0f), AZ::Vector3(1000000.0f, 1000000.0f, 1000000.0f));
-        modelLodCreator.SetMeshAabb(AZStd::move(localAabb));
-
-        RPI::ModelMaterialSlot::StableId slotId = 0;
-        modelCreator.AddMaterialSlot(RPI::ModelMaterialSlot{slotId, AZ::Name{}, AZ::RPI::AssetUtils::LoadAssetByProductPath<AZ::RPI::MaterialAsset>(DefaultSkinnedMeshMaterial)});
-        modelLodCreator.SetMeshMaterialSlot(slotId);
-
-        modelLodCreator.EndMesh();
+        for (uint32_t submeshIndex = 0; submeshIndex < submeshCount; ++submeshIndex)
+        {
+            //
+            // Submesh
+            //
+            modelLodCreator.BeginMesh();
+
+            // Set the index buffer view
+            RHI::BufferViewDescriptor indexBufferDescriptor = indexBuffer->GetBufferViewDescriptor();
+            uint32_t lodTriangleCount = indexBufferDescriptor.m_elementCount / 3;
+            uint32_t meshTriangleCount = lodTriangleCount / submeshCount;
+            uint32_t indexOffset = meshTriangleCount * submeshIndex * 3;
+
+            if (submeshIndex == submeshCount - 1)
+            {
+                meshTriangleCount += lodTriangleCount % submeshCount;
+            }
+            uint32_t indexCount = meshTriangleCount * 3;
+            modelLodCreator.SetMeshIndexBuffer(AZ::RPI::BufferAssetView{ indexBuffer, CreateSubmeshBufferViewDescriptor(indexBuffer, indexCount, indexOffset) });
+
+            // Get the element count and offset for this sub-mesh
+            uint32_t elementCount = verticesPerSubmesh;
+            uint32_t elementOffset = verticesPerSubmesh * submeshIndex;
+
+            // Include any truncated vertices if this is the last mesh
+            if (submeshIndex == submeshCount - 1)
+            {
+                elementCount += totalVertices % verticesPerSubmesh;
+            }
+
+            modelLodCreator.AddMeshStreamBuffer(RHI::ShaderSemantic{ "POSITION" }, AZ::Name(), AZ::RPI::BufferAssetView{ positionBuffer, CreateSubmeshBufferViewDescriptor(positionBuffer, elementCount, elementOffset) });
```

Comment:
nit: We are assuming here (as we do in other places) that first 'layer' will always have position, normal, UVs, tangent and bi-tangent. 
Should we not check for that and work based on the existing data? Even if this is the current state, working based on what exists can help generalizing it for the future.